test/e2e/atlas/identity_providers_test.go

```diff
@@ -0,0 +1,301 @@
+// Copyright 2024 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//go:build e2e || (iam && atlas)
+
+package atlas_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/test/e2e"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20231115010/admin"
+)
+
+func TestIdentityProviders(t *testing.T) {
+	_ = newAtlasE2ETestGenerator(t)
+	req := require.New(t)
+
+	cliPath, err := e2e.AtlasCLIBin()
+	req.NoError(err)
+
+	var federationSettingsID *string
+	var oidcIdentityProviderID *string
+	var orgID string
+	var set bool
+	if orgID, set = os.LookupEnv("MCLI_ORG_ID"); !set {
+		t.Skip("MCLI_ORG_ID must be set")
+	}
```

Comment:
this is true for a lot of tests you don't need to do this nor pass orgid as a flag since it should be able to get it from the env and if not set the command will fail with the orgid not being valid

Reply:
hmmm I removed this and got an error that the response is empty, it seems to not be fetching the orgId from profile, but it works locally

```shell
❯ ./bin/atlas federatedAuthentication federationSettings  describe -P fedauth
ID                         IDENTITY PROVIDER ID   IDENTITY PROVIDER STATUS
6531abc7cbe54754fd77f068   <nil>                  INACTIVE
```

Reply:
nvm me it's working